packages/eslint-plugin-office-addins/src/rules/load-object-before-read.ts

```diff
@@ -0,0 +1,293 @@
+      "getColumnLabelRange",
+      "getDataHierarchy",
+      "getFilterAxisRange",
+      "getPivotItems",
+      "getRowLabelRange",
+      "getDefault",
+      "getPrintArea",
+      "getPrintAreaOrNullObject",
+      "getPrintTitleColumns",
+      "getPrintTitleColumnsOrNullObject",
+      "getPrintTitleRows",
+      "getPrintTitleRowsOrNullObject",
+      "getParentComment",
+      "getAsImage",
+      "getActivePage",
+      "GetStencilInfo",
+      "getBase64Image",
+      "getHtml",
+      "getParagraphInfo",
+      "getByTitle",
+      "getRestApiId",
+      "getByName",
+      "getWindowSize",
+      "getActiveSectionOrNull",
+      "getActiveSection",
+      "getActiveParagraphOrNull",
+      "getActiveParagraph",
+      "getActivePageOrNull",
+      "getActivePage",
+      "getActiveOutlineOrNull",
+      "getActiveOutline",
+      "getActiveNotebookOrNull",
+      "getActiveNotebook",
+      "getCellPadding",
+      "getBorder",
+      "getParagraphBefore",
+      "getParagraphBeforeOrNullObject",
+      "getParagraphAfterOrNullObject",
+      "getParagraphAfter",
+      "getCellOrNullObject",
+      "getHeader",
+      "getFooter",
+      "getTextRanges",
+      "getOoxml",
+      "getNextTextRangeOrNullObject",
+      "getNextTextRange",
+      "getHyperlinkRanges",
+      "getLastOrNullObject",
+      "getDescendants",
+      "getAncestorOrNullObject",
+      "getAncestor",
+      "getByIdOrNullObject",
+      "getById",
+      "getLevelString",
+      "getLevelParagraphs",
+      "getBase64ImageSrc",
+      "getSelection",
+      "getByTypes",
+      "getByTag",
+      "getReferenceId", 
+      "getDocument",
+    ]);
+
+    function findPropertyThatHadToBeLoaded(node: TSESTree.Node | undefined): string {
+      let propertyName = ""; // Will be a string combined with '/' for the case of navigation properties
+      while(node) {
+        if(node.type === TSESTree.AST_NODE_TYPES.MemberExpression
+          && node.property.type === TSESTree.AST_NODE_TYPES.Identifier) {
+          propertyName += node.property.name + "/";
+        }
+        node = node.parent;
+      }
+      return propertyName.slice(0, -1);
+    }
+
+    function isLoadFunction(node: TSESTree.MemberExpression): boolean {
+      return (node.property.type === TSESTree.AST_NODE_TYPES.Identifier
+        && node.property.name === "load");
+    }
+
+    function callsGetAPIFunction(node: TSESTree.Identifier): boolean {
+      return getFunctions.has(node.name);
+    }
+
+    function isGetFunction(node: TSESTree.Expression): boolean {
+      if(node.type == TSESTree.AST_NODE_TYPES.CallExpression
+        && node.callee.type === TSESTree.AST_NODE_TYPES.MemberExpression
+        && node.callee.property.type === TSESTree.AST_NODE_TYPES.Identifier) {
+          if(callsGetAPIFunction(node.callee.property)) {
+            return true;
+          }
+      }
+      return false;
+    }
+
+    function getLoadedPropertyName(node: TSESTree.MemberExpression): string {
```

Comment:
Probably don't need to use "loaded" in the method name here as the body doesn't have any indication of loading anything.  It just retrieves the expression's first argument's name.